src/components/MoneyRequestHeader.tsx

```diff
@@ -199,32 +218,18 @@ function MoneyRequestHeader({
                     shouldShowBackButton={shouldUseNarrowLayout}
                     onBackButtonPress={onBackButtonPress}
                 />
-                {isPending && (
+                {statusBarProps.shouldShow && (
                     <MoneyRequestHeaderStatusBar
                         title={
                             <Icon
-                                src={Expensicons.CreditCardHourglass}
+                                src={statusBarProps.messageIcon}
                                 height={variables.iconSizeSmall}
                                 width={variables.iconSizeSmall}
                                 fill={theme.icon}
                             />
                         }
-                        description={translate('iou.transactionPendingDescription')}
-                        shouldShowBorderBottom={!isScanning}
-                    />
-                )}
-                {isScanning && (
-                    <MoneyRequestHeaderStatusBar
-                        title={
-                            <Icon
-                                src={Expensicons.ReceiptScan}
-                                height={variables.iconSizeSmall}
-                                width={variables.iconSizeSmall}
-                                fill={theme.icon}
-                            />
-                        }
-                        description={translate('iou.receiptScanInProgressDescription')}
-                        shouldShowBorderBottom
+                        description={statusBarProps.messageDescription}
```

Comment:
We can but don't you think it might get less clean? 

Currently all the situations covered by statusBarProps are similar and require almost the same UI design, therefore statusBarProps have quite straightforward type. I like that this makes the code clean and easy to follow.

I see merging the usage would be a good idea if we wanted to remove `PendingMessageProps` type, but because of reasons I mentioned [here](https://github.com/Expensify/App/pull/40354#discussion_r1601461981) I don't think this is what we want to do.